### PR TITLE
Made neighbors field optional in duts.yaml

### PR DESCRIPTION
**Pull request by rewati-arista**
# Please include a summary of the changes

* tests_tools.py -> made presence of neighbors field optional in duts file, so now Vane does not fail if neighbors field is not present in duts file
* test_tests_tools.py -> added tests to verify the above change
* fixture_duts_no_neighbors.yaml -> fixture added for testing wherein duts do not have neighbors

# Include the Issue number and link

https://github.com/aristanetworks/vane/issues/297

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [X] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [ ] Other (please specify)

# Effort required on reviewers end

- - [X] Easy
- - [ ] Medium
- - [ ] Hard 

# How Has This Been Tested?
    
## New feature

Ensured current test cases pass and included newer test cases:

Screenshot 2023-09-26 at 12 37 06 PM

    
# CI pipeline result

- - [X] Pass
- - [ ] Fail

### CODE REVIEW

File: vane/tests_tools.py [hunk 1]
```diff
@@ -269,7 +269,8 @@ def login_duts(test_parameters, test_duts):
         login_ptr["username"] = dut["username"]
         login_ptr["password"] = dut["password"]
         login_ptr["role"] = dut["role"]
-        login_ptr["neighbors"] = dut["neighbors"]
+        if "neighbors" in dut:
```
Comment (kgrozis-arista) [inline vane/tests_tools.py:272]: Would it be better to do a get on neighbors and if it doesn't exist apply an empty dictionary?

Comment (rewati-arista) [reply vane/tests_tools.py:272]: yes we could do:

login_ptr["neighbors"] = dut.get("neighbors", "")

one more clarification, we dont want to create a key neighbors in dut with empty value right? the way we do in the roles PR: https://github.com/aristanetworks/vane/blob/01e4cd2295923511ebdec5f576e7d5e6dce1ae23/vane/tests_tools.py#L271

Comment (kgrozis-arista) [reply vane/tests_tools.py:272]: in hindsight it would be good to be consistent with what we've already done and make it the same as role

**Review by kgrozis-arista — APPROVED** [review]

File: vane/tests_tools.py [hunk 2]
```diff
@@ -269,6 +269,8 @@ def login_duts(test_parameters, test_duts):
         login_ptr["username"] = dut["username"]
         login_ptr["password"] = dut["password"]
         login_ptr["role"] = dut["role"]
+        if "neighbors" not in dut:
+            dut["neighbors"] = ""
         login_ptr["neighbors"] = dut["neighbors"]
```
Comment (grybak-arista) [inline vane/tests_tools.py:274]: As in the other PR, use dictionary.get() instead seems more pythonic
```suggestion
        login_ptr["neighbors"] = dut.get("neighbors", "")
```

**Review by grybak-arista — APPROVED** [review]
Looks good! Just have the same minor suggestion as in the other PR
👍🏻

**Comment by rewati-arista** [comment]
Addressed the merge conflict and PR comment, good to be merged now👍🏻

**Pull request merged**